[tensorexpr] Remove the optional argument in LoopNest::prepareForCodeGen

huiguoo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66967
* #66452
* __->__ #67144
* #67143
* #66451
* #66450



Differential Revision: [D31881150](https://our.internmc.facebook.com/intern/diff/D31881150)

huiguoo committed:
[tensorexpr] Remove the optional argument in LoopNest::prepareForCodeGen

[ghstack-poisoned]

huiguoo committed:
Update on "[tensorexpr] Remove the optional argument in LoopNest::prepareForCodeGen"


Differential Revision: [D31881150](https://our.internmc.facebook.com/intern/diff/D31881150)

[ghstack-poisoned]

huiguoo committed:
Update on "[tensorexpr] Remove the optional argument in LoopNest::prepareForCodeGen"




Differential Revision: [D31881150](https://our.internmc.facebook.com/intern/diff/D31881150)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881150).

ZolotukhinM commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -767,7 +767,7 @@ StmtPtr TensorExprKernel::transformLoops(BackendType backendType, StmtPtr st) {
     auto interm_bufs = l.getIntermediateBufs();
     preAllocIntermediateBufs(interm_bufs);
-    l.prepareForCodegen(interm_bufs);
+    l.prepareForCodegen();
```
Please move `l.prepareForCodegen()` out from the if and remove the `else` branch.

ZolotukhinM reviewed (approved):
Looks good!

huiguoo committed:
Update on "[tensorexpr] Remove the optional argument in LoopNest::prepareForCodeGen"




Differential Revision: [D31881150](https://our.internmc.facebook.com/intern/diff/D31881150)

[ghstack-poisoned]

huiguoo committed:
Update on "[tensorexpr] Remove the optional argument in LoopNest::prepareForCodeGen"




Differential Revision: [D31881150](https://our.internmc.facebook.com/intern/diff/D31881150)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881150).

huiguoo committed:
Update on "[tensorexpr] Remove the optional argument in LoopNest::prepareForCodeGen"




Differential Revision: [D31881150](https://our.internmc.facebook.com/intern/diff/D31881150)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881150).

huiguoo committed:
Update on "[tensorexpr] Remove the optional argument in LoopNest::prepareForCodeGen"




Differential Revision: [D31881150](https://our.internmc.facebook.com/intern/diff/D31881150)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881150).

huiguoo committed:
Update on "[tensorexpr] Remove the optional argument in LoopNest::prepareForCodeGen"




Differential Revision: [D31881150](https://our.internmc.facebook.com/intern/diff/D31881150)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881150).

huiguoo committed:
Update on "[tensorexpr] Remove the optional argument in LoopNest::prepareForCodeGen"




Differential Revision: [D31881150](https://our.internmc.facebook.com/intern/diff/D31881150)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881150).

huiguoo committed:
Update on "[tensorexpr] Remove the optional argument in LoopNest::prepareForCodeGen"




Differential Revision: [D31881150](https://our.internmc.facebook.com/intern/diff/D31881150)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881150).

huiguoo committed:
Update on "[tensorexpr] Remove the optional argument in LoopNest::prepareForCodeGen"




Differential Revision: [D31881150](https://our.internmc.facebook.com/intern/diff/D31881150)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881150).

huiguoo committed:
Update on "[tensorexpr] Remove the optional argument in LoopNest::prepareForCodeGen"




Differential Revision: [D31881150](https://our.internmc.facebook.com/intern/diff/D31881150)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881150).
